Scripts/Interfaces/ISettingsMenuItemProvider.cs

```diff
@@ -0,0 +1,28 @@
+﻿#if UNITY_EDITOR
+using UnityEngine;
+
+namespace UnityEditor.Experimental.EditorVR
+{
+	/// <summary>
+	/// Decorates types that can provide a sub-menu on the Settings menu. The class should also use a MainMenuItemAttribute
```

Comment:
I'm not sure what you mean. What should inherit what?

The MainMenuItemAttribute is also problematic if we continue down the road of one class potentially being on the menu twice.